renderdoc/driver/vulkan/vk_pixelhistory.cpp

```diff
@@ -903,7 +919,40 @@ struct TestsFailedCallback : public VulkanPixelHistoryCallback
   uint32_t GetEventFlags(uint32_t eventId)
   {
     auto it = m_EventFlags.find(eventId);
-    RDCASSERT(it != m_EventFlags.end());
+    if(it == m_EventFlags.end())
+      RDCERR("Can't find event flags for event %u", eventId);
+    return it->second;
+  }
+
+  void FetchOcclusionResults()
+  {
+    if(m_OcclusionQueries.empty())
+      return;
+    m_OcclusionResults.resize(m_OcclusionQueries.size());
+    VkResult vkr =
+        ObjDisp(m_pDriver->GetDev())
+            ->GetQueryPoolResults(
+                Unwrap(m_pDriver->GetDev()), m_OcclusionPool, 0, (uint32_t)m_OcclusionResults.size(),
+                m_OcclusionResults.size() * sizeof(m_OcclusionResults[0]), m_OcclusionResults.data(),
+                sizeof(m_OcclusionResults[0]), VK_QUERY_RESULT_64_BIT | VK_QUERY_RESULT_WAIT_BIT);
+    RDCASSERTEQUAL(vkr, VK_SUCCESS);
+  }
+
+  uint64_t GetOcclusionResult(uint32_t eventId, uint32_t test) const
+  {
+    auto it = m_OcclusionQueries.find(rdcpair<uint32_t, uint32_t>(eventId, test));
+    if(it == m_OcclusionQueries.end())
+      RDCERR("Can't locate occlusion query for event id %u and test flags %u", eventId, test);
+    if(it->second >= m_OcclusionResults.size())
+      RDCERR("Event %u, occlusion index is %u, and the total # of occlusion query data %u", eventId,
```

Comment:
You need `%zu` to print a size_t directly, or else cast the `m_OcclusionResults.size()` parameter to `uint32_t` to match `%u`.

Reply:
Done